lms/djangoapps/certificates/generation_handler.py

```diff
@@ -369,6 +371,15 @@ def _is_cert_downloadable(user, course_key):
     return True
 
 
+def _is_mode_now_eligible(enrollment_mode, cert):
+    """
+    Check if the current enrollment mode is now eligible, while the enrollment mode on the cert is NOT eligible
+    """
+    if modes_api.is_eligible_for_certificate(enrollment_mode) and not modes_api.is_eligible_for_certificate(cert.mode):
```

Comment:
Is this any performance risk in calling `is_eligible_for_certificate` twice in one check? I am looking for this method the code now, but figured it was a reasonable check.

Reply:
I took a look at the code. I no longer have any concern here.